### DISCUSSION TITLE
Problems with Tasmota and Shelly Switch (unwanted reset and remote control question)

### DISCUSSION THREAD

**Discussion by AleXSR700** [arendst/Tasmota#16476] [Q&A] [answered]
Hello everyone,
I thought it would be easy, but it never is ^^

So, I switched to Tasmota and now I have two problems with my Shelly 2.5 which are used as light switches.

1. When the light switch is pressed repeatedly, the device seems to reset. How can I disable this?
2. I want to use one of the two light switches connected to the Shelly 2.5 to control a second Shelly 2.5 (also running Tasmota).

For 2.:
I tried 
`Rule1 ON Power2#state=1 DO http://192.168.0.222/cm?cmnd=Power%20On ENDON`
`Rule2 ON Power2#state=0 DO http://192.168.0.222/cm?cmnd=Power%20Off ENDON`

So when the switch (second input on Shelly 2.5) is turned off, the remote lamp should be turned off. If the switch is turned on, the remote lamp should be turned on.
Is this not correct?

Thank you for your support
Alex
---------------------------------------------------------------------------------------------
EDIT:
There are two questions and multiple answers. Only one can be flagged.

Solution for 1 from @barbudor:     `SetOption1 1`
Solution for 2 from @sfromis :        `Backlog Rule1 ON Power2#state DO publish cmnd/tasmota_device_id/Power0 %value% ENDON; Rule1 1`

**Comment by sfromis** [comment, marked as answer]
Rules **have** to include a command, not just an URL. Preferred strategy is to Publish a command to the other device, instead of using http which is not a feature Tasmota strives to have good support for. If you "insist" on http, the command to use would be `WebSend`.

One rule is enough, Could be something like:
Rule1 ON Power2#state DO publish cmnd/otherdevice/Power %value% ENDON

**Comment by barbudor** [comment]
1. There is a recovery feature that trigger WiFi Manager mode like initial flashing where tasmota opens an access point for configuration. It occurs if you press 6 times on Button_1 and only button 1.
If this is your issue, the only way is to configure the GPIO as Switch 1 with Switchmode 0

2. Did you enabled the rule-set with 'Rule1 1'?
You don't need to use 2 rules set. You can have both rules in Rule1. You can have more than 2000 characters in a single rule-set

**Reply by AleXSR700** [reply]
Thank you @sfromis 
One follow up on 2. How do I define otherdevice? Will this not be via MQTT and rely on a running MQTT server?

The reason I wanted to use http was simply that if wifi is enabled, two shellies can always communicate with each other directly. But if I use MQTT and the server is down (third device) then nothing will happen.

**Reply by barbudor** [reply]
My bad you can disable 6 press with setoption1
See https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions

**Reply by barbudor** [reply]
You can also look at device groups which allow to sync devices

**Reply by sfromis** [reply]
Sure, Publish is via MQTT, but your other thread indicated that you are using MQTT, and thus a broker should be available. Anyway, I just wanted to let you know that using MQTT is *preferable*, and is what Tasmota is designed for. While there is some http support too (the WebSend command I also told you about), is tends to be less reliable in edge cases, and IME a bit slower too.

In my network, the MQTT broker has an uptime in years, with a replacement in case of planned outages, I just consider it an essential service to have operational.

**Reply by AleXSR700** [reply]
Yes, you are correct. I am running an MQTT server. And it is usually turned on. But since it is a Home Assistant setup I do play around with it a bit. It is unlikely, but I do sometimes shut it down.
Is it maybe possible to make a rule depending on availability of MQTT server?

I also seem to still have my http command incorrect:
`Rule1 ON Power2#state DO WebSend http://192.168.0.205/cm?cmnd=Power %value% ENDON`
I get `{"WebSend":"Wrong parameters"}`

Interestingly your cmd, adopted to my case, only turns on Power1 on the other device despite not being specified. Should this not turn on both Power1 and Power2?
`Rule1 ON Power2#state DO publish cmnd/tasmota_2306AE/Power %value% ENDON`

I used a backlog to switch both. Seems like Power is also Power1 on multi relays.
`Rule1 ON Power2#state DO backlog publish cmnd/tasmota_2306AE/Power1 %value%; publish cmnd/tasmota_2306AE/Power2 %value% ENDON`

EDIT:
My switch is behaving strangly. When I turn the hardware switch on, the light turns on. When I turn off, nothing happens. When I turn on again, light turns off.
I set `SwitchMode1 1` and `SwitchMode2 1`
I thought this should fix it, but it seems not.

**Reply by barbudor** [reply]
Your syntax is wrong. Have a look at the doc https://tasmota.github.io/docs/Commands/#websend

Power == Power1
If you want to change all powers of a device it's Power0

I assume you have a rocker switch and want the power to toggle every time you toggle the switch. So the right mode should be switchmode 0
https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode

**Reply by sfromis** [reply]
It helps to *read* the WebSend docs, instead of making a wrong guess about the format, which is more straightforward than using an URL.

Power1 and Power2 are separate triggers and command, they do not work with just one single rule. And I did make the typo of having the command be Power (which indeed defaults to index 1) instead of the correct Power2. The command to act on both Power1 and Power2 at the same time would be `Power0`.

Anyway, I consider it a bad design flaw of Home Assistant to suggest an installation setup tearing down the MQTT broker just because HA had to be cycled. In a good setup, the broker runs independently, as there very rarely is a good case for even considering to close it.

**Reply by AleXSR700** [reply]
I think I simply misread. And it seems that the WebSend command also cannot use the %value% as suggested for MQTT. Because the WebSend command uses On/Off whereas MQTT uses 1/0.

**Reply by barbudor** [reply]
you can use off/on/toggle or 0/1/2 with either http or mqtt. They are fully synonyms

**Reply by AleXSR700** [reply]
> I assume you have a rocker switch and want the power to toggle every time you toggle the switch. So the right mode should be switchmode 0
> https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode

I have a normal light switch. One position is on, one position is off.
And I have tried SwitchMode 0, 1 and 3 and all show the same behaviour.

The device only switches when I return to hardware on.

So, hardware on = on, hardware off = nothing, hardware on = off, hardware off = nothing, hardware on = on.....

**Reply by barbudor** [reply]
Are you sure the GPIO is configured as a switch and not a button?

**Reply by AleXSR700** [reply]
Oh no, I think it might be button
Is: {"NAME":"Shelly 2.5","GPIO":[56,0,19,0,21,127,0,0,6,126,5,22,156],"FLAG":2,"BASE":18}
Should be? {"NAME":"Shelly 2.5","GPIO":[56,0,17,0,21,83,0,0,6,82,5,22,156],"FLAG":2,"BASE":18}

This is embarassing :(

**Reply by barbudor** [reply]
🤣

**Reply by barbudor** [reply]
I don't know the numbers by heart but in the GUI template page you should see switch

**Comment by AleXSR700** [comment]
I can only mark one reply as the answer, so I updated my initial post so people find both answers.

Thank you both for all your support!

And I have to say, the Tasmota Commands page is probably one of the est I have ever seen. I did make a lot of mistakes applying them, but you can basically find almost anything there!
Thank you!
